### PR TITLE
feat: Allow a user to reauthenticate for superuser

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

**Comment by dcramer** [comment]
Maybe we should allow you to simply reauthenticate, but this at least improves the current situation.

**Review by mattrobenolt — APPROVED** [review]

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Changes require @getsentry/security sign-off</td>
    </tr>
  </tbody>
</table>

### Security concerns found

- src/sentry/web/frontend/auth_login.py


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by dcramer** [comment]
#6619

File: src/sentry/web/frontend/auth_login.py [hunk 1]
```diff
@@ -159,7 +159,9 @@ def handle(self, request, *args, **kwargs):
     def get(self, request, *args, **kwargs):
         next_uri = self.get_next_uri(request, *args, **kwargs)
         if request.user.is_authenticated():
-            return self.handle_authenticated(request, *args, **kwargs)
+            # we need to allow re-auth for superusers
+            if not request.user.is_superuser or request.is_superuser():
```
Comment (mitsuhiko) [inline src/sentry/web/frontend/auth_login.py]: this is confusing. Can we add a comment why this is necessary?

**Review by mitsuhiko — DISMISSED** [review]

File: src/sentry/web/frontend/auth_login.py (hunk 1 above)
Comment (dcramer) [reply src/sentry/web/frontend/auth_login.py]: i feel like the comment that i put there gives that, but i expanded it

but now i have to wait another 20m on tests -.-

Comment (mitsuhiko) [reply src/sentry/web/frontend/auth_login.py]: Nevermind. I read `request.is_superuser`.

**Comment by max-wittig** [comment]
Seems like this is not possible with the API or am I missing something?

I can't seem to find a way to access endpoints like `/api/0/users/`

```json
{
  "detail": {
    "message": "You need to re-authenticate for superuser.",
    "code": "superuser-required",
    "extra": {}
  }
}
```

**Pull request merged**